app/src/test/java/nl/tudelft/b_b_w/model/GenesisBlockUnitTest.java

```diff
@@ -0,0 +1,145 @@
+import org.robolectric.annotation.Config;
+
+import nl.tudelft.b_b_w.BuildConfig;
+import nl.tudelft.b_b_w.model.block.BlockData;
+import nl.tudelft.b_b_w.model.block.BlockType;
+import nl.tudelft.b_b_w.model.block.GenesisBlock;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class,sdk= 21,  manifest = "src/main/AndroidManifest.xml")
+public class GenesisBlockUnitTest {
+    private final User a = new User("Alice", "IBANA");
+    private final User b = new User("Bob", "IBANB");
+    private final String PKA = "pka";
+    private final String PKB = "pkb";
+    private final String HASH_GENESIS_A =
+            "7875f4b585ea90f0db3f3fd3efd257a9104147a443877efac3389afe8f26390d";
+
+    private GenesisBlock genesisBlockA;
+    private GenesisBlock genesisBlockA2;
+    private GenesisBlock genesisBlockB;
+
+    @Before
+    public void create() throws HashException {
+        // alice block data
+        BlockData blockDataA = new BlockData();
+        blockDataA.setBlockType(BlockType.GENESIS);
+        blockDataA.setSequenceNumber(1);
+        blockDataA.setTrustValue(TrustValues.INITIALIZED.getValue());
+        blockDataA.setOwner(a);
+        blockDataA.setIban(a);
+        blockDataA.setPreviousHashSender("N/A");
+        blockDataA.setPreviousHashChain("N/A");
+        blockDataA.setPublicKey(PKA);
+        genesisBlockA = new GenesisBlock(blockDataA);
+        genesisBlockA2 = new GenesisBlock(blockDataA);
+
+        // bob block data
+        BlockData blockDataB = new BlockData();
+        blockDataB.setBlockType(BlockType.GENESIS);
+        blockDataB.setSequenceNumber(1);
+        blockDataB.setTrustValue(TrustValues.INITIALIZED.getValue());
+        blockDataB.setOwner(b);
+        blockDataB.setIban(b);
+        blockDataB.setPreviousHashSender("N/A");
+        blockDataB.setPreviousHashChain("N/A");
+        blockDataB.setPublicKey(PKB);
+        genesisBlockB = new GenesisBlock(blockDataB);
+    }
+
+    /** Test hash */
+    @Test
+    public void hash() {
+        assertEquals(HASH_GENESIS_A, genesisBlockA.getOwnHash().toLowerCase());
+    }
+
+    /** Test previous hash sender not available A */
+    @Test
+    public void ownerA() {
+        assertEquals(a, genesisBlockA.getOwner());
+    }
+
+    /** Test previous hash sender not available B */
+    @Test
+    public void ownerB() {
+        assertEquals(b, genesisBlockB.getOwner());
+    }
+
+    /** Test previous hash sender not available */
+    @Test
+    public void previousHashSender() {
+        assertEquals("N/A", genesisBlockA.getPreviousHashSender());
+    }
+
+    /** Test previous hash chain not available */
+    @Test
+    public void previousHashChain() {
+        assertEquals("N/A", genesisBlockA.getPreviousHashChain());
+    }
+
+    /** Test iban */
+    @Test
+    public void iban() {
```

Comment:
don't think 'iban' is a nice name. Change it in something more describing the test. For example 'testIbanGenesis'.